### DISCUSSION TITLE
《Use Headless style to transform antd》

### DISCUSSION THREAD

**Discussion by 1uokun** [ant-design/ant-design-mobile-rn#1395] [General]
[《使用Headless风格改造antd》讨论帖](https://rn.mobile.ant.design/docs/blog/headless-cn)

**Comment by 1uokun** [comment]
## 以 Slider 为例
<table>
<tr>
<th>
业务场景
</th>
<th>
antd设计语言
</th>
<th>
headless风格
</th>
</tr>

<tr>
<td>基础</td>
<td>

```jsx
  <Slider />
```

</td>
<td>

```jsx
<Slider.Root>
    <Slider.Track>
        <Slider.Range />
    </Slider.Tack>
    <Slider.Thumb />
</Slider.Root>
```

</td>
</tr>
<tr>

<td>双滑块按钮</td>

<td>

```jsx
<Slider range/>
```

</td>
<td>

```diff
  <Slider.Root>
      <Slider.Track>
          <Slider.Range />
      </Slider.Tack>
+     <Slider.Thumb index={0}/>
+     <Slider.Thumb index={1}/>
  </Slider.Root>
```

</td>
</tr>

<tr>
<td>轨道样式修改</td>
<td>

```jsx
<Slider styles={{track:{...}}}/>
```

</td>
<td>

```diff
   <Slider.Root>
+     <Slider.Track style={{...}} className={{...}}>
-     <Slider.Track>
          <Slider.Range />
      </Slider.Tack>
      <Slider.Thumb />
   </Slider.Root>
```

</td>
</tr>

<tr>
<td>轨道内添加元素</td>
<td>

不支持❌

</td>
<td>

```diff
   <Slider.Root>
      <Slider.Track>
          <Slider.Range />
+         <View>
+            ...
+         </View>
      </Slider.Tack>
      <Slider.Thumb />
   </Slider.Root>
```

</td>
</tr>
</table>

**Comment by gamebody** [comment]
Headless的问题是比较难用，自定义ui或更精细化的控制需要了解实现细节，作为ant的底层库是比较好的。

**Comment by vbinupx** [comment]
https://gluestack.io/ui/docs/components/all-components
这种形式，包括把每个组件彻底独立出来，这样可自定义的范围就变得很高。主题定制也可以直接通过tailwind实现。确实实现了高度自定义。
